### PR TITLE
Add wboba9 contract

### PR DESCRIPTION
**Pull request by boyuan-chen**
:clipboard: Add associated issues, tickets, docs URL here.

## Overview

Describe what your Pull Request is about in a few sentences.

Add wBOBA9 contract and tests

## Changes

Describe your changes and implementation choices. More details make PRs easier to review.

- Add wBOBA9 contract
- Add deployment file

## Testing

Describe how to test your new feature/bug fix and if possible, a step by step guide on how to demo this.

The unit tests are added

### CODE REVIEW

File: packages/boba/contracts/test/endToEndTests/WBOBA9.spec.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    ;[signer, otherSigner] = await ethers.getSigners()
+  })
+
+  let WBOBA9: Contract
+  beforeEach(async () => {
+    WBOBA9 = await (await ethers.getContractFactory('WBOBA9')).deploy()
+  })
+
+  describe('deposit', () => {
+    it('should create WETH with fallback function', async () => {
```
Comment (InoMurko) [inline packages/boba/contracts/test/endToEndTests/WBOBA9.spec.ts]: WETH?

File: packages/boba/contracts/test/endToEndTests/WBOBA9.spec.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        signer.sendTransaction({
+          to: WBOBA9.address,
+          value: 200,
+        })
+      ).to.not.be.reverted
+
+      expect(await WBOBA9.balanceOf(signer.address)).to.be.equal(200)
+    })
+
+    it('should create WETH with deposit function', async () => {
```
Comment (InoMurko) [inline packages/boba/contracts/test/endToEndTests/WBOBA9.spec.ts]: WETH

File: packages/boba/contracts/test/endToEndTests/WBOBA9.spec.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      await expect(
+        WBOBA9.connect(otherSigner).transferFrom(
+          signer.address,
+          otherSigner.address,
+          50
+        )
+      ).to.be.reverted
+    })
+
+    it('should transfer WETH to an other address when there is approvement', async () => {
```
Comment (InoMurko) [inline packages/boba/contracts/test/endToEndTests/WBOBA9.spec.ts]: weth

**Review by InoMurko — CHANGES_REQUESTED** [review]

File: packages/boba/contracts/test/endToEndTests/WBOBA9.spec.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      expect(await WBOBA9.balanceOf(signer.address)).to.be.equal(100)
+    })
+  })
+
+  describe('withdraw', () => {
+    it('should revert when withdraw amount is bigger than balance', async () => {
+      await expect(WBOBA9.withdraw(10000)).to.be.reverted
+    })
+
+    it('should withdraw to eth', async () => {
```
Comment (InoMurko) [inline packages/boba/contracts/test/endToEndTests/WBOBA9.spec.ts]: Eth 😆

**Review by InoMurko — CHANGES_REQUESTED** [review]

**Review by InoMurko — APPROVED** [review]
Scan the files for ETH

File: packages/boba/contracts/test/endToEndTests/WBOBA9.spec.ts [hunk 5]
```diff
@@ -0,0 +1,98 @@
+import chai, { expect } from 'chai'
+import chaiAsPromised from 'chai-as-promised'
+chai.use(chaiAsPromised)
+import { ethers } from 'hardhat'
+import { Contract } from 'ethers'
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers'
+
+describe('WBOBA9', () => {
```
Comment (InoMurko) [inline packages/boba/contracts/test/endToEndTests/WBOBA9.spec.ts:8]: This does look like very redundant testing of a standard erc20?

Comment (Boyuan-Chen) [reply packages/boba/contracts/test/endToEndTests/WBOBA9.spec.ts:8]: I added the WETH9 test.

**Review by wsdt — APPROVED** [review]

**Review by souradeep-das — APPROVED** [review]

**Review by mmontour1306 — COMMENTED** [review]
Looks OK to me

**Pull request merged**